src/edu/csus/ecs/pc2/shadow/ShadowController.java

```diff
@@ -605,9 +605,11 @@ private Map<String, String> getRemoteSubmissionsToJudgementsMap() {
      * Returns a Map which maps submissionIds to {@link ShadowJudgementInfo} objects for every submission (run) in the
      * specified array of PC2 Runs.
      */
-    private Map<String, ShadowJudgementInfo> getJudgementsMap(Run[] pc2Runs) {
+    protected Map<String, ShadowJudgementInfo> getJudgementsMap(Run[] pc2Runs) {
 
         Map<String, ShadowJudgementInfo> pc2JudgementInfoMap = new HashMap<String, ShadowJudgementInfo>();
+        
+        log = getLog();
```

Comment:
This call is already made in method `start()`.    It doesn't seem like a good idea to have multiple places throughout the code doing assignments to `log`.    (What probably should instead be done is to replace every reference to variable `log` with a call to `getLog()`...)